src/components/MealCostCalculator.vue

```diff
@@ -0,0 +1,304 @@
+                {{option.text}}
+              </option>
+            </select></td>
+            <td><strong>{{calculateNetValue(quantity, price, ingredientIndex)}}</strong></td>
+            <td><strong>{{calculateGrossValue(quantity, price, vat, ingredientIndex)}}</strong></td>
+          </tr>
+        </tbody>
+      </table>
+      <div class="newIngredient">
+        <input placeholder="Ingredient name" @keypress.enter="addIngredient()"  @input="handleInput" :value="value">
+        <button class="dim button" @click="addIngredient()">Add new ingredient</button>
+      </div>
+      <table class="weightTable" border = "1">
+        <thead>
+          <th>Total weight/Kg:</th>
+          <th>Portion weight/g:</th>
+          <th>Number of servings:</th>
+        </thead>
+        <tbody>
+          <td><input type="number" :id="`totalWeight`" placeholder="Edit" v-model.number="meal.portions.totalWeight"></td>
+          <td><input type="number" :id="`portionWeight`" placeholder="Edit" v-model.number="meal.portions.portionWeight"></td>
+          <td>{{calculatePortionsCount(meal.portions.totalWeight, meal.portions.portionWeight)}}</td>
+        </tbody>
+      </table>
+      <table class="costTable" border = "1">
+        <thead>
+          <th></th>
+          <th>Net cost:</th>
+          <th>Gross cost:</th>
+        </thead>
+        <tbody>
+          <tr>
+            <td><strong>Total:</strong></td>
+            <td>{{totalNetCost()}}</td>
+            <td>{{totalGrossCost()}}</td>
+          </tr>
+          <tr>
+            <td><strong>Per portion:</strong></td>
+            <td>{{portionNetCost()}}</td>
+            <td>{{portionGrossCost()}}</td>
+          </tr>
+        </tbody>
+      </table>
+    <button class="dim button">SAVE</button>
+  </div>
+</template>
+
+<script>
+import { uuid } from 'vue-uuid'; 
+export default {
+  name: "MealCostCalculator",
+   data() {
+    return {
+      meal: {
+        name:"Bigos",
+        id:1,
+        ingredients:[
+          {
+          id:1,
+          name:"Lopatka",
+          quantity:4.1,
+          price:10.45,
+          vat:"1", 
+          gross:0,
+          net:0,
+          },
+          {
+          id:2,
+          name:"Kielbasa",
+          quantity:1.6,
+          price:20.99,
+          vat:"1",
+          gross:0,
+          net:0,
+          },
+        ],
+        portions: {
+          totalWeight: "",
+          portionWeight: "",
+          portionsCount:"",
+          portionNetCost: 0,
+          portionGrossCost: 0,
+        },
+        totalNetCost: 0,
+        totalGrossCost: 0,
+        totalPortionNetCost: 0,
+        totalportionGrossCost: 0,
+      },
+
+      vatOptions: [
+        { text: "Select Vat", value:1, disabled: true},
+        { text: "5", value:1.05},
+        { text: "8", value:1.08},
+        { text: "23", value:1.23}      
+      ],
+      value: "",
```

Comment:
Rename to `addIngredientInputValue` or smh descriptive

Reply:
✔️